chains/evm/evmtransaction/evm-tx.go

```diff
@@ -31,15 +29,38 @@ func (a *TX) RawWithSignature(key *ecdsa.PrivateKey, chainID *big.Int) ([]byte,
 		return nil, err
 	}
 	a.tx = tx
-	rawTX, err := rlp.EncodeToBytes(tx)
+
+	data, err := tx.MarshalBinary()
 	if err != nil {
 		return nil, err
 	}
-	return rawTX, nil
 
+	return data, nil
+}
+
+func NewTransaction(chainId *big.Int, nonce uint64, to *common.Address, amount *big.Int, gasLimit uint64, gasPrice *big.Int, gasTipCap *big.Int, gasFeeCap *big.Int, data []byte) evmclient.CommonTransaction {
```

Comment:
Again why we are not using some GasPricer? YOu can pass it as argument.
I see its interface as 
type  GasPricer interface {
   DynamicFee() bool // does it supports dynamic fees
   GasPrice(client) [2]big.Int //if DynamicFee() == true then it returns 2 args, if false 1 arg
}
And 2 base implementations  with DefaultGasprice from config + Dynamic one

Reply:
I made a default implementation that returns whichever gas price it is passed, but I was unsure if this is what you actually wanted or instead wanted one that just returns the literal value from the config file.

Reply:
I also made it to check for the second item of the array returned from `GasPrice` inside the transaction constructor to determine whether it is an EIP 1559 tx. Also, as the gasPricer has access to a client they can determine a dynamic fee tx by getting the `BaseFee`